typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -341,15 +330,36 @@ export abstract class HyperlaneDeployer<
     return implementation.attach(proxy.address) as C;
   }
 
-  private cacheContract<K extends keyof Factories>(
+  private writeCache<K extends keyof Factories>(
     chain: ChainName,
     contractName: K,
-    contract: HyperlaneContracts<Factories>[K],
+    address: types.Address,
   ) {
-    if (!this.deployedContracts[chain]) {
-      this.deployedContracts[chain] = {} as HyperlaneContracts<Factories>;
+    if (!this.cachedAddresses[chain]) {
+      this.cachedAddresses[chain] = {};
     }
-    this.deployedContracts[chain][contractName] = contract;
+    this.cachedAddresses[chain][contractName] = address;
+  }
+
+  private readCache<F extends ethers.ContractFactory>(
+    chain: ChainName,
+    factory: F,
+    contractName: string,
+  ): { hit: boolean; contract: Awaited<ReturnType<F['deploy']>> } {
+    const cachedAddress = this.cachedAddresses[chain]?.[contractName];
+    const hit = !!cachedAddress;
+    const contractAddress = hit ? cachedAddress : ethers.constants.AddressZero;
+    const contract = factory
+      .attach(contractAddress)
+      .connect(this.multiProvider.getSignerOrProvider(chain)) as Awaited<
+      ReturnType<F['deploy']>
+    >;
+    if (hit) {
+      this.logger(
+        `Recovered ${contractName.toString()} on ${chain} ${cachedAddress}`,
+      );
+    }
+    return { hit, contract };
```

Comment:
Why is a `hit` value useful instead of just checking if a `contract` value does or does not exist in the cache?

Reply:
It's not, we can just return contract | undefined